src/accept/types.ts

```diff
@@ -27,6 +27,10 @@ type VisitorWithGenericNodeType = {
 };
 
 type VisitorWithSpecificNodeTypes = {
-  [ts.SyntaxKind.InterfaceDeclaration]?: VisitSyntaxKindFunction<ts.InterfaceDeclaration>;
+  [ts.SyntaxKind.InterfaceDeclaration]?: VisitSyntaxKindFunction<InterfaceDeclaration>;
   [ts.SyntaxKind.ArrayType]?: VisitSyntaxKindFunction<ts.ArrayTypeNode>;
 };
+
+interface InterfaceDeclaration extends Omit<ts.InterfaceDeclaration, 'members'> {
```

Comment:
A new interface is declared, overwriting the original native `ts.InterfaceDeclaration` type. By casting the node `as any` we can do this and the compiler will pass our new custom type `InterfaceDeclaration` to the syntax kind specific visitor function.

This new type specifies the property `members` as a union type, thus allowing type narrowing via the `kind` property.

Of course this type narrowing is not complete but it serves as an example.